engine/classes/Elgg/Application/CacheHandler.php

```diff
@@ -199,19 +203,59 @@ protected function sendCacheHeaders($etag) {
 	}
 
 	/**
-	 * Send content type
+	 * Get the content type
 	 *
 	 * @param string $view The view name
-	 * @return void
+	 * 
+	 * @return string?
 	 */
-	protected function sendContentType($view) {
+	protected function getContentType($view) {
+		
+		$extension = (new \SplFileInfo($view))->getExtension();
+		
+		switch ($extension) {
+			case 'gif':
+			case 'png':
+			case 'webp':
+			case 'bmp':
+			case 'tiff':
+				return "image/$extension";
+			
+			case 'jpeg':
+			case 'jpg':
+				return "image/jpeg";
+
+			case 'ico':
+				return 'image/x-icon';
+			
+			case 'svg':
+				return 'image/svg+xml';
+			
+			case 'js':
+				return 'application/javascript';
+			
+			case 'css':
+				return 'text/css';
+				
+			case 'html':
+				return 'text/html';
+				
+			case 'xml':
+				return 'text/xml';
+			
+			default:
+				break;
```

Comment:
We'll want to make this extensible probably, but for now I just hard-coded a bunch of common asset media types.